code/messaging/src/phone/java/com/adobe/marketing/mobile/InAppNotificationHandler.java

```diff
@@ -49,204 +57,89 @@ class InAppNotificationHandler {
-     * by the Optimize extension for retrieving offers
-     *
-     * @param activityId  a {@code String} containing the activity id
-     * @param placementId a {@code String} containing the placement id
-     * @return a base64 encoded JSON string to be used by the Optimize extension for retrieving offers
-     */
-    static String getEncodedDecisionScopeForActivityAndPlacement(final String activityId, final String placementId) {
-        final byte[] decisionScopeBytes = String.format("{\"%s\":\"%s\",\"%s\":\"%s\",\"itemCount\":%s}",
-                MessagingConstants.EventDataKeys.Optimize.ACTIVITY_ID,
-                activityId,
-                MessagingConstants.EventDataKeys.Optimize.PLACEMENT_ID,
-                placementId,
-                MessagingConstants.DefaultValues.Optimize.MAX_ITEM_COUNT)
-                .getBytes(StandardCharsets.UTF_8);
-
-        final EncodingService encodingService = MobileCore.getCore().eventHub.getPlatformServices().getEncodingService();
-        return new String(encodingService.base64Encode(decisionScopeBytes));
-    }
-
-    /**
-     * Takes the retrieved application identifier and returns a base64 encoded JSON string in the format expected
-     * by the Optimize extension for retrieving offers
-     *
-     * @param applicationId a {@code String} containing the application id
-     * @return a base64 encoded JSON string to be used by the Optimize extension for retrieving offers
-     */
-    static String getEncodedDecisionScopeForAppId(final String applicationId) {
-        final byte[] decisionScopeBytes = String.format("{\"%s\":\"%s\"}", MessagingConstants.EventDataKeys.Optimize.XDM_NAME, applicationId).getBytes(StandardCharsets.UTF_8);
-
-        final EncodingService encodingService = MobileCore.getCore().eventHub.getPlatformServices().getEncodingService();
-        return new String(encodingService.base64Encode(decisionScopeBytes));
-    }
-
-    /**
-     * Handles the Offers notification payload by extracting the rules json objects present in the Edge response event payload
+     * Handles the notification payload by extracting the rules json objects present in the Edge response event
      * into a list of {@code Map} objects. The valid rule json objects are then registered in the {@link RulesEngine}.
      *
-     * @param payload A {@link Map<String, Variant>} containing the personalization decision payload retrieved via the Optimize extension.
+     * @param payload A {@link Map<String, Variant>} containing the personalization decision payload retrieved via the Edge extension.
      */
-    void handleOfferNotificationPayload(final Map<String, Variant> payload) {
+    void handlePersonalizationPayload(final Map<String, Object> payload) {
         if (MessagingUtils.isMapNullOrEmpty(payload)) {
             Log.warning(LOG_TAG, "%s - Empty content returned in call to retrieve in-app messages.", SELF_TAG);
```

Comment:
maybe this should be a debug instead of a warn?

Reply:
and it should also let the developer know that you're clearing cache as a result :D

Reply:
makes sense, i'll update